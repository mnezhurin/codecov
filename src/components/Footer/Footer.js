import React from 'react';

export const Footer = ({ classNames = '', children }) => {
    return (
        <main className={classNames}>
            {children}
        </main>
    );
};
