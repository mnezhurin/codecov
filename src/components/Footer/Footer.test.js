import React from 'react';
import renderer from 'react-test-renderer';
import {Footer} from './Footer';
import Link from "../Link/Link";

const LIST_OF_LINKS = [
    { to: 'google.com', text: 'Google'},
    { to: 'facebook.com', text: 'Facebook'},
    { to: '/news', text: 'Our news'},
];

test('Footer structure', () => {
    const component = renderer.create(
        <Footer><ul>
            {LIST_OF_LINKS.map(({ text, to}, index) => <Link key={index} page={to} classNames="App-link" children={text} />)}
        </ul></Footer>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});