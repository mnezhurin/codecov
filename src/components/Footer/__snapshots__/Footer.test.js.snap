// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer structure 1`] = `
<main
  className=""
>
  <ul>
    <a
      className="App-link normal "
      href="google.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Google
    </a>
    <a
      className="App-link normal "
      href="facebook.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Facebook
    </a>
    <a
      className="App-link normal "
      href="/news"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Our news
    </a>
  </ul>
</main>
`;
